admin/services/check-start.service.js

```diff
@@ -99,14 +99,34 @@ checkStartService.pupilLogin = async function (pupilId) {
     throw new Error('Unable to find a prepared check for pupil: ' + pupilId)
   }
 
-  const res = await checkFormDataService.sqlGetActiveForm(check.checkForm_id)
-  if (!res) {
+  let checkForm = null
+  let checkForms = []
+
+  // If they have not logged in before, then give selected form.
+
+  if (check.pupilLoginDate === null) {
+    const res = await checkFormDataService.sqlGetActiveForm(check.checkForm_id)
+
+    checkForm = res ? R.head(res) : null
+  } else {
+    const allForms = await checkFormService.getAllFormsForCheckWindow(check.checkWindow_id)
+
+    checkForms = JSON.parse(`[${check.checkForm_ids}]`)
```

Comment:
If checkForm_ids is not defined JSON.parse will throw.
I'd recommend to safely handle a case like this.